workerapi/server/server.go

```diff
@@ -21,99 +24,99 @@ func Serve(handler worker.Worker, addr string, transportFactory thrift.TTranspor
 	processor := worker.NewWorkerProcessor(handler)
 	server := thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
 
-	fmt.Println("About to serve")
+	fmt.Println("Serving thrift: ", addr)
 
 	return server.Serve()
 }
 
-type Handler struct {
-	runs map[string]*worker.RunStatus
-	stat stats.StatsReceiver
+type handler struct {
+	stat            stats.StatsReceiver
+	run             runner.Runner
+	getVersion      func() string
+	timeLastContact time.Time
 }
 
-func NewHandler(stat stats.StatsReceiver) worker.Worker {
+func DefaultHandler(stat stats.StatsReceiver) worker.Worker {
 	if stat == nil {
 		stat = stats.NilStatsReceiver()
 	}
+	return NewHandler(
+		stat,
+		localrunner.NewSimpleRunner(execer.NewExecer()),
+		func() string { return "" },
+	)
+}
+
+func NewHandler(stat stats.StatsReceiver, run runner.Runner, getVersion func() string) worker.Worker {
 	scopedStat := stat.Scope("handler")
-	return &Handler{make(map[string]*worker.RunStatus), scopedStat}
+	h := &handler{stat: scopedStat, run: run, getVersion: getVersion}
+	go h.stats()
+	return h
 }
 
-func (h *Handler) QueryWorker() (*worker.WorkerStatus, error) {
-	wstatus := worker.NewWorkerStatus()
-	wstatus.Running = []string{}
-	for runId, status := range h.runs {
-		if status.Status == worker.Status_RUNNING {
-			wstatus.Running = append(wstatus.Running, runId)
+// Periodically output stats
+func (h *handler) stats() {
+	ticker := time.NewTicker(time.Millisecond * time.Duration(500))
+	for {
+		select {
+		case <-ticker.C:
```

Comment:
Agree why not update the stats next to the code they are tracking thats a more common pattern.